Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM quay.io/centos/centos:stream9 as builder
+
+WORKDIR /retis
+
+RUN dnf install -y \
+    libpcap-devel \ 
+    clang \
+    llvm \ 
+    rustfmt \
+    cargo \
+    elfutils-libelf-devel \
+    zlib-devel 
+
+# Only download the dependencies for now so these steps can be cached.
+RUN cargo init
+COPY Cargo.lock .
+COPY Cargo.toml .
+RUN cargo fetch --locked
+
+# Now copy the rest of the source and build.
+COPY . .
```

Comment:
I thought of adding it but discarded the idea because:
- It's not expected to be run on a dirty trees (it's essentially a github hook).
- I didn't want to add a maintenance burden. For starters we would also need to add `build.rs`. If we implement some preliminary buildings steps that requires scripts or other tooling outside `src` (e.g: getting kernel headers) or the future configuration sample files (i.e: `profiles`) would also need to be added. Also shell autocompleters, etc